engine/src/main/java/org/terasology/math/MatrixUtils.java

```diff
@@ -245,4 +247,10 @@ public static Matrix3f calcNormalMatrix(Matrix4f mv) {
         result.transpose();
         return result;
     }
+
+    public static Quat4f extractRotation(BaseMatrix4f m) {
```

Comment:
It might be good here to add a docstring explaining the formulas. They are somewhat dense, and not obvious to the uninitiated. Is there some upstream documentation for these formulas?